docs/adr/0002-use-mask-as-from-address.md

```diff
@@ -0,0 +1,200 @@
+From: "offers@servicex.example.com [via Relay]" offers@servicex.example.com
+To: users-real-email@mail.example.com
+Reply-To: replies@relay.firefox.com
+Resent-From: r4nd0m@mozmail.com
+Resent-Sender: replies@relay.firefox.com
+```
+
+This format was inspired by the
+[ReSender example](https://datatracker.ietf.org/doc/html/rfc5598#section-5.2) from
+[RFC 5598: Internet Mail Architecture](https://datatracker.ietf.org/doc/html/rfc5598).
+It used standards-based headers to make it clear who sent the message and who
+resent it.
+
+- Good, because the sender's email address is available for filtering.
+- Bad, because [DKIM](https://en.wikipedia.org/wiki/DomainKeys_Identified_Mail)
+  verification caused messages to be marked as spam or dropped entirely.
+
+### Option 3: Use the Relay Mask as the `From:` Address
+
+If the sender's email headers are:
+
+```
+Subject: A special offer for you
+From: "Your friends at service X" <offers@servicex.example.com>
+To: r4nd0m@mozmail.com
+```
+
+Relay forwards with:
+
+```
+Subject: A special offer for you
+From: "offers@servicex.example.com [via Relay]" r4nd0m@mozmail.com
+To: users-real-email@mail.example.com
+Reply-To: replies@relay.firefox.com
+Resent-From: offers@servicex.example.com
+```
+
+- Good, because the original sender is visible in the display name.
+- Good, because the email mask is available for filtering.
+- Good, because Proton Mail organizes by mask in conversation mode.
+- Good, because it rewards usage of a mask on a single service, which is better
+  for user privacy and is compatible with the Relay business model.
+- Bad, because of a small increased probability of delivery to spam, due to the
+  reputation of `mozmail.com` versus `mozilla.com`.
+
+In QA testing in August 2023, emails from one service were sent to spam with
+the new `From:` address, while were delivered to the inbox with the old `From:`
+address from Option 1, while over 100 saw no difference in Inbox delivery.
+
+### Option 4: Use the Relay Mask Plus Derived Prefix as the `From:` Address
+
+If the sender's email headers are:
+
+```
+Subject: A special offer for you
+From: "Your friends at service X" <offers@servicex.example.com>
+To: r4nd0m@mozmail.com
+```
+
+Relay forwards with:
+
+```
+Subject: A special offer for you
+From: "offers@servicex.example.com [via Relay]" r4nd0m+offers_at_servicex.example.com@mozmail.com
+To: users-real-email@mail.example.com
+Reply-To: replies@relay.firefox.com
+Resent-From: offers@servicex.example.com
+```
+
+- Good, because the original sender is visible in the display name.
+- Good, because the email mask plus original email is available for filtering.
+- Good, because Proton Mail organizes by mask in conversation mode.
+- Bad, because of a small increased probability of delivery to spam, due to the
+  reputation of `mozmail.com` versus `mozilla.com`.
+- Bad, because it encourages reuse of masks, which is worse for user privacy
+  and for the Relay business model.
```

Comment:
```suggestion
- Bad, because of a small increased probability of delivery to spam, due to the
  reputation of `mozmail.com` versus `firefox.com`.
- Bad, because it encourages reuse of masks, which is worse for user privacy.
```